draft-janfred-radext-radiusdtls-bis.md

```diff
@@ -183,26 +183,24 @@ While the client may be able to deduce the operational state of the local server
 Within RADIUS, proxies typically only forward traffic between the NAS and RADIUS servers, and they do not generate their own response.
 As a result, when a NAS does not receive a response to a request, this could be the result of packet loss between the NAS and proxy, a problem on the proxy, loss between the RADIUS proxy and server, or a problem with the server.
 
-When UDP is used as a transport protocol, the absence of a reply can cause a client to deduce (incorrectly) that the proxy is unavailable.
+The absence of a reply can cause a client to deduce (incorrectly) that the proxy is unavailable.
 The client could then fail over to another server or conclude that no "live" servers are available (OKAY state in {{!RFC3539}}, Appendix A).
 This situation is made even worse when requests are sent through a proxy to multiple destinations.
 Failures in one destination may result in service outages for other destinations, if the client erroneously believes that the proxy is unresponsive.
 
-For RADIUS/TLS, it is RECOMMENDED that implementations utilize the existence of a TCP connection along with the application-layer watchdog defined in {{RFC3539}}, Section 3.4 to determine that the server is "live".
-RADIUS/TLS clients MUST mark a connection DOWN, if the network stack indicates that the connection is no longer active.
-If the network stack indicates that the connection is still active, clients MUST NOT decide that it is down until the application-layer watchdog algorithm has marked it DOWN.
-RADIUS/TLS clients MUST NOT decide that a RADIUS/TLS server is unresponsive until all TLS connections to it have been marked down.[^contradiction]{:jf}
+It is REQUIRED that implementations utilize the existence of a TCP/DTLS connection along with the application-layer watchdog defined in {{RFC3539}}, Section 3.4 to determine that the server liveliness.
+RADIUS/TLS clients MUST mark a connection DOWN, under the following conditions:
+- The administrator has marked the connection administrative DOWN.
+- The network stack indicates that the connection is no longer viable.
+- The application-layer watchdog algorithm has marked it DOWN.
 
-[^contradiction]: The specification in RFC6613 is contradictory here. Section 2.4 says that it is recommended to have a watchdog. Section 2.6 says it must be used.
+For RADIUS/TLS, TCP connections MAY send TCP keepalives as described in {{RFC9293}} Section 3.8.4, that UDP connections MAY send periodic DTLS keepalives as defined in {{RFC6520}}, as a way of proactively and rapidly triggering a connection DOWN notification from the network stack.  These liveliness checks are essentially redundant in the presence of an application-layer watchdog, but may provide more rapid notifications of connectivity issues.
```

Comment:
TCP and DTLS level keepalives are redundant.  DTLS heartbeats aren't available in OpenSSL (they were moved).  The behaviour of the application-layer watchdog is superior in that it only runs in the _absence_ of normal request/response traffic, and does not result in spurious keepalive/heartbeart packets.  Transport level keepalives/heartbeats primarily useful because the minimum interval of Status-Server packets is 6 seconds.  keepalives and heartbeats may be sent more frequently and could provide more timely indications of connectivity loss.